tests/phpunit/tests/blocks/renderHooks.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Tests for the hooks of the block rendering functions.
+ *
+ * @package WordPress
+ * @subpackage Blocks
+ * @since 6.4.0
+ *
+ * @group blocks
+ */
+class Tests_Blocks_Render_Hooks extends WP_UnitTestCase {
+	/**
+	 * @ticket 59131
+	 */
+	public function test_do_blocks_pre_render_filter() {
+		$block_content = '
+			<!-- wp:core/paragraph --><p>Hello</p><!-- /wp:core/paragraph -->
+			<!-- wp:core/paragraph --><p>WordPress!</p><!-- /wp:core/paragraph -->
+		';
+
+		add_filter( 'do_blocks_pre_render', array( $this, 'do_blocks_pre_render_filter' ) );
+
+		$result = do_blocks( $block_content );
+
+		$this->assertSame( '<p>Hello</p>', trim( $result ) );
+
+		remove_filter( 'do_blocks_pre_render', array( $this, 'do_blocks_pre_render_filter' ) );
```

Comment:
Minor thing, but the filter could be removed before the assertion so it gets executed even when the assertion fails. This increases the chances it won't leak into other test cases causing random issues.

Reply:
Makes sense, thanks. Done in https://github.com/WordPress/wordpress-develop/pull/5023/commits/69cbad7d5a3176169712b4396d08ec6a723339ad.